### ISSUE TITLE
Rule `leading-zero` misfiring

### ISSUE THREAD

**Issue by shnhrrsn** [sasstools/sass-lint#882]
**What version of Sass Lint are you using?**
1.9.1

**Please include any relevant parts of your configuration**
```yaml
options:
  formatter: stylish
  merge-default-rules: false
rules:
  hex-notation:
    - 1
    - style: lowercase
  leading-zero: true
```

**What did you do? Please include the actual source code causing the issue.**

Run the following file `test.scss` via `sass-lint test.scss --verbose`:

```scss
div {
  color: rgba(#FFF, 0.5);
}
```

**What did you expect to happen?**

An issue reported for `hex-notation` and nothing more.

**What actually happened? Please include any error messages given to you by Sass Lint.**

```bash
$ sass-lint test.scss --verbose

test.scss
  2:15  warning  Hex notation should all be lower case   hex-notation
  2:21  warning  Don't include leading zeros on numbers  leading-zero
```

If you satisfy the `hex-notation` rule, changing `#FFF` to `#fff`, running `sass-lint test.scss --verbose` will report no issues.

However, if you satisfy the `hex-notation` rule, but then change `0.5` to `.5` as follows:

```scss
div {
  color: rgba(#fff, .5);
}
```

The linter will still report no issues, despite the fact that the leading zero is missing on `.5`.

**Comment by shnhrrsn** [comment]
Misunderstood how the rule worked, apologies!

When configured as:

```yml
  leading-zero:
    - 1
    - include: true
```

All is well.
